src/components/LabEditorGUI/SwitchNode.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import switchSVG from '../../assets/icons/switch.svg';
+import longSwitchSVG from '../../assets/icons/switch-48port.svg';
+import changeSelected from '../../redux/actions/changeGUI';
+import {useDispatch, useSelector} from 'react-redux';
+
+const SwitchNode = ({id, inputs, outputs}: any) => {
+
+  const selectedNode = useSelector((state: any) => state.gui);
+  const dispatch = useDispatch();
+
+  const handleSelect = () => {
+    // Timeout will override the deselect click event in the diagram div
+    // There might be a better way to handle this
+    setTimeout(() => dispatch(changeSelected({selectedID: id})), 10);
+  };
+
+  function generatePorts(portType: any) {
+    return(
+    portType.map((port: any) => React.cloneElement(port, {
```

Comment:
Can you explain why you are using React.cloneElement here? Did this library recommend this?

Reply:
Yes, this is the syntax the library uses when it explains customization and dynamic nodes.